matchzoo/engine/base_generator.py

```diff
@@ -0,0 +1,81 @@
+"""Base generator."""
+
+import abc
+import keras
+import numpy as np
+
+
+class BaseGenerator(keras.utils.Sequence):
+    """Abstract base generator of all matchzoo generators.
+
+    Every `Generator` must implement the `_get_batch_of_transformed_samples`
```

Comment:
Every `Generator` must implement the :meth:\`_get_batch_of_transformed_samples\`

`:meth:` represents for a method, just like `:class:` and `:attr:` is used to generate docstrings.